src/Signature/SignatureTrait.php

```diff
@@ -0,0 +1,37 @@
+<?php
+namespace Aws\Signature;
+
+/**
+ * Provides signature calculation for SignatureV4.
+ */
+trait SignatureTrait
+{
+    public function getSignatureV4(
```

Comment:
`public` and `protected` methods should be avoided for anything that is called from within the class. Methods should be private by default (especially in a trait).